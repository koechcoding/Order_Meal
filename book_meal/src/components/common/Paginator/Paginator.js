import React from "react";
import PropTypes from "prop-types";
import { Button } from 'reactstrap';

const Paginator = ({ pageInfo = {}, onNext, onPrev}) =>(
    <div>
    <p className="text-center pt-0 mt-0 text-bottom">Showing page {pageInfo.currentPage} of {pageInfo.pageCount}</p>
    <div className="pt-1">
        <Button 
            disabled={!pageInfo.hasNext} 
            onClick={() => onNext(pageInfo.nextPage)} 
            className="btn btn-secondary float-right btn-next">Next </Button>
        <Button 
            disabled={!pageInfo.hasPrev} 
            onClick={() => onPrev(pageInfo.prevPage)} 
            className="btn btn-secondary btn-prev"> Previous</Button>
    </div>
</div>
);

Paginator.propTypes = {
    pageInfo: PropTypes.object,
    onNext: PropTypes.func.isRequired,
    onPrev: PropTypes.func.isRequired,
}

export default Paginator;