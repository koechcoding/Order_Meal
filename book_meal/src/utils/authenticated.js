export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}