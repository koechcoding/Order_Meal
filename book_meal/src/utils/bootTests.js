import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
