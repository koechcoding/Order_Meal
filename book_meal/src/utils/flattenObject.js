export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}