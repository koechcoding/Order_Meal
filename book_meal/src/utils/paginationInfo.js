export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}