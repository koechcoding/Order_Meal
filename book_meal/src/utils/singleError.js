const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;