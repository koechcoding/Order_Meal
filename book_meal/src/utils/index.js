import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}