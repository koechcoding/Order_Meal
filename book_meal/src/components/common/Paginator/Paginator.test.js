import React from 'react';
import { shallow }  from 'enzyme';
import sinon from 'sinon';
import 'src/utils/bootTests'; // order matters!
import Paginator from './Paginator';
import Adapter from 'enzyme-adapter-react-16';

let pageInfo ={
    hasNext: true,
    hasPrev: true,
    prevPage: 1,
    nextPage: 2,
};

describe ('<Paginator/>', ()=>{
    it('renders correctly', () => {
        const wrapper = shallow(<Paginator pageInfo={pageInfo} onNext={() => {}} onPrev={() => {}} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('shows next button', () => {
        const spy = sinon.fake();
        const wrapper = shallow(<Paginator onNext={spy} onPrev={() => {}} pageInfo={pageInfo}/>);
        wrapper.find('.btn-next').simulate('click');
        expect(spy.called).toBe(true);
    });
    it('shows previous button', () => {
        const spy = sinon.fake();
        const wrapper = shallow(<Paginator onPrev={spy} onNext={() => {}} pageInfo={pageInfo}/>);
        wrapper.find('.btn-prev').simulate('click');
        expect(spy.called).toBe(true);
    });
});