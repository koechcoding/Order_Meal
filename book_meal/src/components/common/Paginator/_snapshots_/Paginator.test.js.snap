// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paginator
    onNext={[Function]}
    onPrev={[Function]}
    pageInfo={
        Object {
            "hasNext": true,
            "hasPrev": true,
            "nextPage": 2,
            "prevPage": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="text-center pt-0 mt-0 text-bottom"
>
          Showing page 
           of 
</p>,
        <div
          className="pt-1"
>
          <Button
                    className="btn btn-secondary float-right btn-next"
                    color="secondary"
                    disabled={false}
                    onClick={[Function]}
                    tag="button"
          >
                    Next 
          </Button>
          <Button
                    className="btn btn-secondary btn-prev"
                    color="secondary"
                    disabled={false}
                    onClick={[Function]}
                    tag="button"
          >
                     Previous
          </Button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Showing page ",
            undefined,
            " of ",
            undefined,
          ],
          "className": "text-center pt-0 mt-0 text-bottom",
        },
        "ref": null,
        "rendered": Array [
          "Showing page ",
          undefined,
          " of ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="btn btn-secondary float-right btn-next"
              color="secondary"
              disabled={false}
              onClick={[Function]}
              tag="button"
>
              Next 
</Button>,
            <Button
              className="btn btn-secondary btn-prev"
              color="secondary"
              disabled={false}
              onClick={[Function]}
              tag="button"
>
               Previous
</Button>,
          ],
          "className": "pt-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Next ",
              "className": "btn btn-secondary float-right btn-next",
              "color": "secondary",
              "disabled": false,
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": "Next ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Previous",
              "className": "btn btn-secondary btn-prev",
              "color": "secondary",
              "disabled": false,
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": " Previous",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="text-center pt-0 mt-0 text-bottom"
>
            Showing page 
             of 
</p>,
          <div
            className="pt-1"
>
            <Button
                        className="btn btn-secondary float-right btn-next"
                        color="secondary"
                        disabled={false}
                        onClick={[Function]}
                        tag="button"
            >
                        Next 
            </Button>
            <Button
                        className="btn btn-secondary btn-prev"
                        color="secondary"
                        disabled={false}
                        onClick={[Function]}
                        tag="button"
            >
                         Previous
            </Button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Showing page ",
              undefined,
              " of ",
              undefined,
            ],
            "className": "text-center pt-0 mt-0 text-bottom",
          },
          "ref": null,
          "rendered": Array [
            "Showing page ",
            undefined,
            " of ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="btn btn-secondary float-right btn-next"
                color="secondary"
                disabled={false}
                onClick={[Function]}
                tag="button"
>
                Next 
</Button>,
              <Button
                className="btn btn-secondary btn-prev"
                color="secondary"
                disabled={false}
                onClick={[Function]}
                tag="button"
>
                 Previous
</Button>,
            ],
            "className": "pt-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Next ",
                "className": "btn btn-secondary float-right btn-next",
                "color": "secondary",
                "disabled": false,
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": "Next ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Previous",
                "className": "btn btn-secondary btn-prev",
                "color": "secondary",
                "disabled": false,
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": " Previous",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;